test/Microsoft.ML.NightlyBuild.Tests/Microsoft.ML.NightlyBuild.Tests.csproj

```diff
@@ -6,6 +6,8 @@
     <!-- DON'T CHANGE THIS!!! We are turning off strong naming to ensure we never add `InternalsVisibleTo` for these tests -->
     <PublicSign>false</PublicSign>
     <IsPackable>false</IsPackable>
+    <!-- NU1605 is because the dependencies are automatically updated via a script at runtime -->
+    <NoWarn>$(NoWarn);NU1605</NoWarn>
```

Comment:
Why do we need this?  We shouldn't be downgrading dependencies.

Reply:
the nightly build uses a script to update the dependencies to the latest version, so the versions listed in the file aren't used. See `NightlyBuildDependency.props`. Is there a better way to go about saying that? That we don't need to worry about the specific versions in that file?

Reply:
Just delete the direct reference if it's coming in transitively with a higher version. 

I just looked at the checked in process here for nightly builds and it seems pretty hacky.  A better solution would be a separate build leg that can consume packages just built, and knows their version and can pass that in as a parameter.  I wonder if a) we're even running this anymore and b) anyone is looking at the failures?  I don't see any runs: https://dev.azure.com/dnceng-public/public/_build?definitionId=169